### DISCUSSION TITLE
Interacting - flick a lever.

### DISCUSSION THREAD

**Discussion by j-romchain** [ORelio/Minecraft-Console-Client#1744] [Q&A] [answered]
I want to have it start a farm, to do so it just needs to flick the lever. Are there commands to 1. Find the lever, and 2.flick the lever?

**Comment by ORelio** [comment, marked as answer]
~~You would need to write a C# script, but I'm not sure if there is a way to retrieve the state of the lever currently.
Example of script: [sample-script-with-world-access.cs](https://github.com/ORelio/Minecraft-Console-Client/blob/master/MinecraftClient/config/sample-script-with-world-access.cs) (lamp state can be retrieved in 1.12 and lower at the moment)~~ OK I misread your question, you want to flip the lever, not watch for it and run an action in MCC, see answer below.

You might be able to do so using the [`/useblock`](https://github.com/ORelio/Minecraft-Console-Client/blob/master/MinecraftClient/Commands/Useblock.cs) command, however I'm not sure you can look for a certain block type at the moment, you would need to open the game first and take not of the exact coordinates.

**Reply by j-romchain** [reply]
Actually you can search for exact block states, see your example lamp watcher.

**Reply by j-romchain** [reply]
Could you add a command to tell the blocks of interest around you or a specified block?

**Reply by ORelio** [reply]
Block states are not collected past 1.13 due to the way terrain is sent now.
For a "block search" feature, feel free to open a feature request by creating an Issue 🙂

**Reply by j-romchain** [reply]
So lamp watcher no longer works?

**Reply by ORelio** [reply]
Unfortunately, yes

**Reply by j-romchain** [reply]
Can you fix it?

**Reply by ORelio** [reply]
Not easily as Minecraft now has more than 17000 different block states. Related: #599
